sql/_13_issues/_24_1h/answers/cbrd_25277.answer

```diff
@@ -0,0 +1,221 @@
+0
+===================================================
+0
+===================================================
+1
+===================================================
+1
+===================================================
+1
+===================================================
+1
+===================================================
+1
+===================================================
+0
+===================================================
+0
+===================================================
+0
+===================================================
+test case 1    
+AS-IS - Outer join query optimization failed.
+TO-BE
+- Eliminate INNER JOIN for the first table
+- INNER JOIN
+- With ANSI JOIN syntax
+     
+
+Query plan:
+sscan
+    class: dual node[?]
+    cost:  ? card ?
+Query stmt:
+select 'AS-IS - Outer join query optimization failed.
+===================================================
+c_b    c_b    
+2     20     
+
+Query plan:
+idx-join (inner join)
+    outer: sscan
+               class: c node[?]
+               cost:  ? card ?
+    inner: iscan
+               class: d node[?]
+               index: pk_tc_dummy_c_a term[?]
+               cost:  ? card ?
+    cost:  ? card ?
+Query stmt:
+select /*+ ORDERED */ c.c_b, d.c_b from  inner join tb_child c inner join tc_dummy d on d.c_a=c.c_a
+===================================================
+test case 2    
+AS-IS - No error with ORACLE JOIN syntax.
+TO-BE
+- Eliminate INNER JOIN for the first table
+- INNER JOIN
+- With ORACLE JOIN syntax
+     
+
+Query plan:
+sscan
+    class: dual node[?]
+    cost:  ? card ?
+Query stmt:
+select 'AS-IS - No error with ORACLE JOIN syntax.
+===================================================
+c_b    c_b    
+2     20     
+
+Query plan:
+idx-join (inner join)
+    outer: sscan
+               class: c node[?]
+               cost:  ? card ?
+    inner: iscan
+               class: d node[?]
+               index: pk_tc_dummy_c_a term[?]
+               cost:  ? card ?
+    cost:  ? card ?
+Query stmt:
+select /*+ ORDERED */ c.c_b, d.c_b from  inner join tb_child c, tc_dummy d where d.c_a=c.c_a
+===================================================
+test case 3    
+AS-IS - Outer join query optimization failed.
+TO-BE
+- Eliminate INNER JOIN for the first table
+- LEFT OUTER JOIN
+- With ANSI JOIN syntax
+     
+
+Query plan:
+sscan
+    class: dual node[?]
+    cost:  ? card ?
+Query stmt:
+select 'AS-IS - Outer join query optimization failed.
+===================================================
+c_b    c_b    
+2     20     
+3     null     
+
+Query plan:
+idx-join (left outer join)
+    outer: sscan
+               class: c node[?]
+               cost:  ? card ?
+    inner: iscan
+               class: d node[?]
+               index: pk_tc_dummy_c_a term[?]
+               cost:  ? card ?
+    cost:  ? card ?
+Query stmt:
+select /*+ ORDERED */ c.c_b, d.c_b from  inner join tb_child c left outer join tc_dummy d on d.c_a=c.c_a
```

Comment:
This is the same as the review written above.

Reply:
I hope this will be discussed in another issue.

Reply:
The issue of needing to remove the 'inner join' keyword will be resolved in the CBRD-25323 issue.